File: Server/src/main/java/com/api/server/foodb/CiteThisTextbookReferencing.java
package com.api.server.foodb;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

import java.time.Instant;

@Getter
@Setter
@Entity
@Table(name = "cite_this_textbook_referencings")
public class CiteThisTextbookReferencing {
    @Id
    @Column(name = "id", nullable = false)
    private Integer id;

    @NotNull
    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @OnDelete(action = OnDeleteAction.CASCADE)
    @JoinColumn(name = "textbook_id", nullable = false)
    private CiteThisTextbook textbook;

    @NotNull
    @Column(name = "referencer_id", nullable = false)
    private Integer referencerId;

    @Size(max = 255)
    @NotNull
    @Column(name = "referencer_type", nullable = false)
    private String referencerType;

    @NotNull
    @Column(name = "created_at", nullable = false)
    private Instant createdAt;

    @NotNull
    @Column(name = "updated_at", nullable = false)
    private Instant updatedAt;

    @Size(max = 255)
    @Column(name = "pages")
    private String pages;

    @Size(max = 255)
    @Column(name = "chapter")
    private String chapter;

    @Size(max = 255)
    @Column(name = "chapter_author")
    private String chapterAuthor;

}

File: DotNetServer/DotNetServer/Core/Entities/Foodb/CiteThisTextbookReferencing.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace DotNetServer.Core.Entities.Foodb;

[Table("cite_this_textbook_referencings")]
[Index("ReferencerId", "ReferencerType", Name = "cite_this_article_referencings_all_ids")]
[Index("ReferencerId", Name = "index_cite_this_textbook_referencings_on_referencer_id")]
[Index("TextbookId", Name = "index_cite_this_textbook_referencings_on_textbook_id")]
public class CiteThisTextbookReferencing
{
    [Key] [Column("id")] public int Id { get; set; }

    [Column("textbook_id")] public int TextbookId { get; set; }

    [Column("referencer_id")] public int ReferencerId { get; set; }

    [Column("referencer_type")] public string ReferencerType { get; set; } = null!;

    [Column("created_at", TypeName = "datetime")]
    public DateTime CreatedAt { get; set; }

    [Column("updated_at", TypeName = "datetime")]
    public DateTime UpdatedAt { get; set; }

    [Column("pages")] [StringLength(255)] public string? Pages { get; set; }

    [Column("chapter")]
    [StringLength(255)]
    public string? Chapter { get; set; }

    [Column("chapter_author")]
    [StringLength(255)]
    public string? ChapterAuthor { get; set; }

    [ForeignKey("TextbookId")]
    [InverseProperty("CiteThisTextbookReferencings")]
    public virtual CiteThisTextbook Textbook { get; set; } = null!;
}